crates/core/src/db/commit_log.rs

```diff
@@ -121,3 +127,150 @@ impl CommitLog {
         }
     }
 }
+
+/// A read-only view of a [`CommitLog`].
+pub struct CommitLogView {
+    mlog: Option<Arc<Mutex<MessageLog>>>,
```

Comment:
I might suggest implementing the `CommitLog` by giving it a `CommitLogView` internally, since it does already share the same fields. Hmm that doesn't seem like the right solution either, but I wonder if we can have a clearer abstraction here.

It seems like we are almost trying to create the concept of a "File" (i.e. `CommitLogView`) and a "BufferedFile" (i.e. `CommitLog`) where the buffered thing is unwritten commit which is only periodically flushed. I wonder if something like that might fit better down the road.

Reply:
The purpose of `CommitLogView` is for `RelationalDB` to hand out a view over its _internal state_ in order to serve up log segments. It must hide `append`, but changing the visibility of `append` isn't helpful.

`CommitLogView` could be thought of as a newtype around `CommitLog`, but internally that'd be copying more data than strictly required. Perhaps if there was an `Arc` around the whole `CommitLog`, but I'm always wary about contention.